From 54a8e3f74af1a01f62d017f6d368cff5244c3437 Mon Sep 17 00:00:00 2001
From: Dinesh Karthikeyan <quic_dinek@quicinc.com>
Date: Tue, 9 Aug 2022 11:23:20 +0530
Subject: [PATCH] ath12k: Account correct segment count in coreudmp

Calculate the number fo segments based on the number of segments
allocated in qmi.

Signed-off-by: Dinesh Karthikeyan <quic_dinek@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/coredump.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

Index: backports-20220404-5.4.164-f40abb4788/drivers/net/wireless/ath/ath12k/coredump.c
===================================================================
--- backports-20220404-5.4.164-f40abb4788.orig/drivers/net/wireless/ath/ath12k/coredump.c	2022-08-11 12:42:59.417750740 +0530
+++ backports-20220404-5.4.164-f40abb4788/drivers/net/wireless/ath/ath12k/coredump.c	2022-08-11 12:44:09.741916507 +0530
@@ -163,7 +163,8 @@ void ath12k_coredump_download_rddm(struc
 	for (i = 0; i < ab->qmi.mem_seg_count; i++) {
 		if (ab->qmi.target_mem[i].type == HOST_DDR_REGION_TYPE ||
 		    ab->qmi.target_mem[i].type == CALDB_MEM_REGION_TYPE ||
-		    ab->qmi.target_mem[i].type == M3_DUMP_REGION_TYPE)
+		    ab->qmi.target_mem[i].type == M3_DUMP_REGION_TYPE ||
+		    ab->qmi.target_mem[i].type == PAGEABLE_MEM_TYPE)
 			rem_seg_cnt++;
 	}
 
@@ -219,6 +220,7 @@ void ath12k_coredump_download_rddm(struc
 			seg_info->vaddr = ab->qmi.target_mem[i].v.ioaddr;
 			seg_info->type = ATH12K_FW_PAGEABLE_DATA;
 			seg_info++;
+			continue;
 		}
 		if (ab->qmi.target_mem[i].type == M3_DUMP_REGION_TYPE) {
 			seg_info->len = ab->qmi.target_mem[i].size;
@@ -226,6 +228,7 @@ void ath12k_coredump_download_rddm(struc
 			seg_info->vaddr = ab->qmi.target_mem[i].v.ioaddr;
 			seg_info->type = ATH12K_FW_M3_DUMP;
 			seg_info++;
+			continue;
 		}
 		seg_info->len = ab->qmi.target_mem[i].size;
 		seg_info->addr = ab->qmi.target_mem[i].paddr;
